code/modules/mob/living/simple_animal/slime/slime.dm

```diff
@@ -494,66 +369,78 @@
-	Friends[new_friend] = amount
-	if(new_friend)
-		RegisterSignal(new_friend, COMSIG_QDELETING, PROC_REF(clear_memories_of), override = TRUE)
-
-///Removes someone from the friendlist
-/mob/living/simple_animal/slime/proc/remove_friend(friend)
-	Friends -= friend
-	if(friend && !SLIME_CARES_ABOUT(friend))
-		UnregisterSignal(friend, COMSIG_QDELETING)
-
-///Adds someone to the friend list
-/mob/living/simple_animal/slime/proc/set_friends(new_buds)
-	clear_friends()
-	for(var/mob/friend as anything in new_buds)
-		set_friendship(friend, new_buds[friend])
+///Handles slime attacking restrictions, and any extra effects that would trigger
+/mob/living/simple_animal/slime/proc/slime_pre_attack(mob/living/simple_animal/slime/our_slime, atom/target, proximity, modifiers)
+	SIGNAL_HANDLER
+	if(isAI(target)) //The aI is not tasty!
+		to_chat(our_slime, "This being is incompatible, and not tasty.")
+		return COMPONENT_CANCEL_ATTACK_CHAIN
+
+	if(our_slime.buckled == target) //If you try to attack the creature you are latched on, you instead cancel feeding
+		our_slime.stop_feeding()
+		return COMPONENT_CANCEL_ATTACK_CHAIN
+
+	if(iscyborg(target))
+		var/mob/living/silicon/robot/borg_target = target
+		borg_target.flash_act()
+		do_sparks(5, TRUE, borg_target)
+		var/stunprob = our_slime.powerlevel * 7 + 10
+		if(prob(stunprob) && our_slime.powerlevel >= 8)
+			our_slime.powerlevel -= 3
+			borg_target.adjustBruteLoss(our_slime.powerlevel * rand(6,10))
+			borg_target.visible_message(span_danger("The [our_slime.name] shocks [borg_target]!"), span_userdanger("The [our_slime.name] shocks you!"))
+		else
+			borg_target.visible_message(span_danger("The [our_slime.name] fails to hurt [borg_target]!"), span_userdanger("The [our_slime.name] failed to hurt you!"))
 
-///Removes everyone from the friend list
-/mob/living/simple_animal/slime/proc/clear_friends()
-	for(var/mob/friend as anything in Friends)
-		remove_friend(friend)
+		return COMPONENT_CANCEL_ATTACK_CHAIN
 
-///The passed source will be no longer be the slime's target, leader, or one of its friends
-/mob/living/simple_animal/slime/proc/clear_memories_of(datum/source)
-	SIGNAL_HANDLER
-	if(source == Target)
-		set_target(null)
-	if(source == Leader)
-		set_leader(null)
-	remove_friend(source)
+	if(iscarbon(target) && our_slime.powerlevel > 0)
+		var/mob/living/carbon/carbon_target = target
+		var/stunprob = our_slime.powerlevel * 7 + 10  // 17 at level 1, 80 at level 10
+		if(!prob(stunprob))
+			return
 
-#undef SLIME_CARES_ABOUT
+		carbon_target.visible_message(span_danger("The [our_slime.name] shocks [carbon_target]!"), span_userdanger("The [our_slime.name] shocks you!"))
+
+		do_sparks(5, TRUE, carbon_target)
+		var/power = our_slime.powerlevel + rand(0,3)
+		carbon_target.Paralyze(power * 2 SECONDS)
+		carbon_target.set_stutter_if_lower(power * 2 SECONDS)
+		if (prob(stunprob) && our_slime.powerlevel >= 8)
+			carbon_target.adjustFireLoss(our_slime.powerlevel * rand(6,10))
```

Comment:
Suggestion: `apply_damage` instead
```suggestion
			carbon_target.apply_damage(our_slime.powerlevel * rand(6, 10), BURN, spread_damage = TRUE, wound_bonus = CANT_WOUND)
```

Reply:
Good idea, both the spreading it all over the mob, and preventing double wounding.